File: backend/src/main/java/com/codecool/training_portal/dto/group/project/questionnaire/QuestionnaireUpdateRequestDto.java
package com.codecool.training_portal.dto.group.project.questionnaire;

import com.codecool.training_portal.model.group.project.questionnaire.QuestionnaireStatus;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import org.hibernate.validator.constraints.Length;

import java.util.Set;

public record QuestionnaireUpdateRequestDto(
  @NotNull @Length(min = 1, max = 100) String name, @NotNull String description,
  @NotNull QuestionnaireStatus status,
  @NotNull Set<@Valid QuestionCreateRequestDto> questions) {
}

File: frontend/src/questionnaires/dto/QuestionnaireUpdateRequestDto.ts
import {QuestionCreateRequestDto} from "./QuestionCreateRequestDto.ts";
import {QuestionnaireStatus} from "./QuestionnaireStatus.ts";

export interface QuestionnaireUpdateRequestDto {
  name: string;
  description: string;
  status: QuestionnaireStatus;
  questions: QuestionCreateRequestDto[];
}
